### PR TITLE
Update metrics docs

### LINKED ISSUE #7639: Add docs for stashing metrics
Add some docs for the new metrics added in #7257.

### PR DESCRIPTION
**Pull request by billygriffin**
Closes #7639

## Description
This adds documentation to the metrics.md file that describes the metrics. These were already added to the marketing site, so this is largely to catch up with that: https://desktop.github.com/usage-data/

Notes: No notes

### CODE REVIEW

**Comment by Rexogamer** [comment]
one thing about the periods: shouldn’t it be GitHub.com still?

**Comment by tierninho** [comment]
thanks @Rexogamer I think I got them all!

**Comment by outofambit** [comment]
lol my bad 🤦🏼‍♀️ thanks @Rexogamer and @tierninho !

**Comment by Rexogamer** [comment]
final nitpick: shouldn’t the same go for pullrebase? (it was pull.rebase originally)

File: docs/process/metrics.md [hunk 1]
```diff
@@ -29,58 +29,74 @@ These are general metrics about feature usage and specific feature behaviors. Th
...
+| `mergeSuccessAfterConflictsCount` | The number of times the user successfully completes a merge after a merge conflict | To understand how effectively users are able to resolve conflicts and complete their merge successfully |
+| `noActionTakenOnStashCount` | The number of times the user views a stash but neither restores nor discards it | To understand whether people might find value in just seeing the stash without acting on it |
+| `openShellCount` | The number of times the user has opened a shell from the app | To understand if people need to use the command line because of missing features |
+| `partialCommits` | The number of partial commits | To understand usage patterns of commits made in Desktop |
+| `prBranchCheckouts` | The number of times the user checks out a branch using the PR menu | To understand usage patterns around the PR checkout menu |
+| `pullWithDefaultSettingCount` | The number of times a user has pulled with `pullrebase` unset or set to `false` | To understand how many users have "pull with rebase" enabled as part of their workflow |
```
Comment (outofambit) [inline docs/process/metrics.md]:
```suggestion
| `pullWithDefaultSettingCount` | The number of times a user has pulled with `pull.rebase` unset or set to `false` | To understand how many users have "pull with rebase" enabled as part of their workflow |
```

File: docs/process/metrics.md [hunk 2]
```diff
@@ -29,58 +29,74 @@ These are general metrics about feature usage and specific feature behaviors. Th
...
+| `pullWithRebaseCount` |  The number of times a user performed a pull with `pullrebase` in config set to `true` | To understand how many users have "pull with rebase" enabled as part of their workflow |
```
Comment (outofambit) [inline docs/process/metrics.md]:
```suggestion
| `pullWithRebaseCount` |  The number of times a user performed a pull with `pull.rebase` in config set to `true` | To understand how many users have "pull with rebase" enabled as part of their workflow |
```

**Pull request merged**